lua/neo-tree/sources/diagnostics/lib/items.lua

```diff
@@ -105,7 +105,7 @@ M.get_diagnostics = function(state)
       if diag_items_by_buffer[bufnr] == nil then
         diag_items_by_buffer[bufnr] = {}
       end
-      table.insert(diag_items_by_buffer[bufnr], diag_item)
+      diag_items_by_buffer[bufnr][diag_item.id] = diag_item
```

Comment:
This is indeed much cleaner :)

Reply:
only if `table.sort` recreates numerical indexes though

Reply:
yeah it doesn't work `table.sort` just ignores values with a non-numerical index

Reply:
This could work too (and I agree, it's pretty clean), but you would then have to convert `diag_items` to an array-like table before it is sorted and passed to `item.children` (lines 134-135), since `diag_items` has now been changed to the format `{ <id> = <item>, ... }`. You could probably use `vim.tbl_values` for this. 

I think the set approach is more readable and is probably a bit easier to maintain. It is somewhat more verbose though.

Reply:
Hadn't thought about reading neovim's manual when searching for such an obvious standard library function lol

Yeah the intent behind the set can't be missed. I'll push that and let you decide :)

Reply:
Looks great :)